app/main.py

```diff
@@ -16,4 +16,30 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+def sorted_variables_func(*args):
+    types = {"mutable" : ["list", "dict", "set"],
+             "immutable" : ["int", "float", "str", "bool", "tuple"]}
+
+    variables_dict = {key : value for key, value in globals().items()
+                      if not key.startswith("__")}
+
+    mutable = []
+    immutable = []
+
+    for arg in args:
+        var_name = [key for key, value in variables_dict.items()
+                    if value is arg]
```

Comment:
The list comprehension is used to find the variable name corresponding to the argument. This works, but it assumes that the variable values are unique, which might not always be the case. If two variables have the same value, this could lead to unexpected results.